source/base/tensor_product_polynomials.cc

```diff
@@ -510,9 +511,6 @@ TensorProductPolynomials<dim, PolynomialType>::evaluate(
   std::vector<Tensor<3, dim>> &third_derivatives,
   std::vector<Tensor<4, dim>> &fourth_derivatives) const
 {
-  if constexpr (dim == 0)
-    return;
-
```

Comment:
Would the warning have gone away if you had put the rest into an `else` branch?

Reply:
Out of curiosity: Do you dislike the change because you would not have liked to see the code written in that way to begin with, or because you don't like patches that change indentation? I believe that we should write patches that reflect the code we want, instead of so that they change the minimal number of lines. In other words, our code base should reflect our best effort to produce good and readable code, not reflect its history.

Reply:
I prefer if the code does not indent too much irrespective history, which is why I wrote the early return (I know you prefer the `if() ... else ...` form). Upon your suggestion, I then realized that the code has many of those template specialization, which I didn't want to touch all. Generally, if I'm given the alternative to indent more or to have separate functions, I would often go with separate functions if the code can be logically split like here. I would not be opposed to switch all occurrences in this class to `if constexpr (dim > 0) {...}`, because here the indentation is not worrisome. That said, I am certainly guilty of using `if (...) return;` in some of my code because it can lead to simpler control flow or less indentation when there are multiple conditions, but I also see the point in your preference, so I try to use it only in selected cases.

Reply:
I think that's a case of personal preference, and I'll leave it at your discretion :-)